### PR TITLE
Network Check for location history and overlap screen.

### PR DESCRIPTION
**Pull request by Krish023**
Network Check for location history and overlap screen.
Cache enable for news section if users are offline #262

### CODE REVIEW

File: app/views/News.js [hunk 1]
```diff
@@ -33,11 +33,13 @@ class NewsScreen extends Component {
       name: 'Safe Paths News', // TODO: translate
       url: 'https://privatekit.mit.edu',
     };
+
     this.state = {
       visible: true,
       default_news: default_news,
       newsUrls: [default_news, default_news],
       current_page: 0,
+      iscached: false,
```
Comment (tstirrat) [inline app/views/News.js:42]: follow other key_name_convention. Also, `is_cached` is unclear which cache it's referring to. 

Perhaps `enable_webview_cache` is more clear?

File: app/views/News.js [hunk 2]
```diff
@@ -56,6 +58,11 @@ class NewsScreen extends Component {
     });
   }
 
+  handleConnectionInfoChange = () => {
+    NetInfo.fetch().then(state => {
+      if (!state.isConnected) this.setState({ iscached: true });
+    });
+  };
```
Comment (tstirrat) [inline app/views/News.js:65]: add a blank line between methods

File: app/locales/en/english.json [hunk 3]
```diff
@@ -185,6 +185,7 @@
     "tested_positive_subtitle": "Your private data can be transferred to health authorities, backed up, or otherwise shared.",
 
     "choose_provider_title": "Choose health authority",
-    "choose_provider_subtitle": "To be informed of exposures you will need to subscribe to a health authority."
+    "choose_provider_subtitle": "To be informed of exposures you will need to subscribe to a health authority.",
+    "offlineMessage": "Please check your internet connection and try again."
```
Comment (tstirrat) [inline app/locales/en/english.json:189]: key naming consistency: `"offline_message"`

File: app/views/LocationTracking.js [hunk 4]
```diff
@@ -288,7 +294,9 @@ class LocationTracking extends Component {
   }
 
   overlap() {
-    this.props.navigation.navigate('OverlapScreen', {});
+    if (this.state.isUserOffline) {
+      alert(languages.t('offlineMessage'));
```
Comment (tstirrat) [inline app/views/LocationTracking.js:298]: Add a test

File: app/views/Settings.js [hunk 5]
```diff
@@ -68,7 +73,9 @@ class SettingsScreen extends Component {
   }
 
   eventHistoryButtonPressed() {
-    this.props.navigation.navigate('NotificationScreen');
+    if (this.state.isUserOffline) {
+      alert(languages.t('offlineMessage'));
```
Comment (tstirrat) [inline app/views/Settings.js:77]: Add a test for this

**Comment by tstirrat** [comment]
Please commit the `yarn.lock` change too, this will fix the build

File: app/views/News.js (hunk 1 above)
Comment (murari85b7) [reply app/views/News.js]: Changed variable name

File: app/views/News.js (hunk 2 above)
Comment (murari85b7) [reply app/views/News.js:67]: added new line

File: app/locales/en/english.json (hunk 3 above)
Comment (murari85b7) [reply app/locales/en/english.json]: Changed key name

File: app/views/LocationTracking.js (hunk 4 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: add a test means unit test case?

File: app/views/Settings.js (hunk 5 above)
Comment (murari85b7) [reply app/views/Settings.js]: add a test means unit test case?

File: app/views/LocationTracking.js (hunk 4 above)
Comment (tstirrat) [reply app/views/LocationTracking.js]: Yes please. LocationTracking.spec.js and also for Settings

File: app/views/Settings.js (hunk 5 above)
Comment (tstirrat) [reply app/views/Settings.js]: Yes please, I will add base Settings.spec.js

**Comment by tstirrat** [comment]
Resync to get the Settings base test in #406 406

**Comment by kenpugsley** [comment]
Update on this ... we're trying to lock the changed for v1, we hope to branch to master for v1 very soon.  After that, we should come back to review/merge this.  @Krish023, we will probably need you to resolve the conflicts at some point so we can consider this right after the branch for v1 to master.

**Pull request closed without merging**